### PR TITLE
Add DFN-8 for Linear LTC2451

### PR DESCRIPTION
**Pull request by yankee14**
Add 8-pin DFN with 1 EP from Linear LTC2451

Datasheet: https://www.analog.com/media/en/technical-documentation/data-sheets/2451fg.pdf#page=17

kicad-symbols: https://github.com/KiCad/kicad-symbols/pull/2791
kicad-footprints: https://github.com/KiCad/kicad-footprints/pull/2312
kicad-footprint-generator: https://github.com/pointhi/kicad-footprint-generator/pull/569

Screencap:

DFN-8-1EP_2x3mm_P0 5mm_EP0 56x2

image

### CODE REVIEW

**Comment by yankee14** [comment]
Ignore git branch name, I misread the datasheet when I branched.

**Comment by yankee14** [comment]
The silkscreen line is confusing, it suggests pin 1 is on the top right. But the assembly layer shows it correctly.

File: scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/dfn.yaml [hunk 1]
```diff
@@ -261,6 +261,50 @@ DFN-8-1EP_2x2mm_P0.5mm_EP0.7x1.3mm:
...
+DFN-8-1EP_3x2mm_P0.5mm_EP0.56x2.15mm:
+  device_type: 'DFN'
+  #manufacturer: 'man'
+  #part_number: 'mpn'
+  size_source: 'https://www.analog.com/media/en/technical-documentation/data-sheets/2451fg.pdf#page=17'
+  # ipc_class: 'qfn' # 'qfn_pull_back'
+  #ipc_density: 'least' #overwrite global value for this device.
+  body_size_x:
+    nominal: 2
+    tolerance: 0
```
Comment (chschlue) [inline scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/dfn.yaml:273]: 0.1

File: scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/dfn.yaml [hunk 2]
```diff
@@ -261,6 +261,50 @@ DFN-8-1EP_2x2mm_P0.5mm_EP0.7x1.3mm:
...
+  body_size_y:
+    nominal: 3
+    tolerance: 0
```
Comment (chschlue) [inline scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/dfn.yaml:276]: 0.1

File: scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/dfn.yaml [hunk 3]
```diff
@@ -261,6 +261,50 @@ DFN-8-1EP_2x2mm_P0.5mm_EP0.7x1.3mm:
...
+  overall_height:
+    minimum: 0.7
+    maximum: 0.8
```
Comment (chschlue) [inline scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/dfn.yaml:279]: Please use nominal and tolerance, as given in the datasheet.

File: scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/dfn.yaml [hunk 4]
```diff
@@ -261,6 +261,50 @@ DFN-8-1EP_2x2mm_P0.5mm_EP0.7x1.3mm:
...
+  lead_width:
+    minimum: 0.2
+    maximum: 0.3
```
Comment (chschlue) [inline scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/dfn.yaml:282]: See above

File: scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/dfn.yaml [hunk 5]
```diff
@@ -261,6 +261,50 @@ DFN-8-1EP_2x2mm_P0.5mm_EP0.7x1.3mm:
...
+  lead_len:
+    minimum: 0.3
+    maximum: 0.5
```
Comment (chschlue) [inline scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/dfn.yaml:285]: See above

File: scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/dfn.yaml [hunk 6]
```diff
@@ -261,6 +261,50 @@ DFN-8-1EP_2x2mm_P0.5mm_EP0.7x1.3mm:
...
+
+  EP_size_x:
+    minimum: 0.51
+    maximum: 0.61
```
Comment (chschlue) [inline scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/dfn.yaml:289]: See above

File: scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/dfn.yaml [hunk 7]
```diff
@@ -261,6 +261,50 @@ DFN-8-1EP_2x2mm_P0.5mm_EP0.7x1.3mm:
...
+# EP_size_x_overwrite: 0.56
+  EP_size_y:
+    minimum: 2.1
+    maximum: 2.2
```
Comment (chschlue) [inline scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/dfn.yaml:293]: See above

**Review by chschlue — CHANGES_REQUESTED** [review]

**Comment by yankee14** [comment]
Hmm I have never seen this change request feature before. Is it like applying patch files? This is cool.

File: scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/dfn.yaml (hunk 1 above)
Comment (yankee14) [reply scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/dfn.yaml:273]: What do you mean by 0.1? The tolerance?

Comment (yankee14) [reply scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/dfn.yaml:273]: Yes you must mean that :)

**Review by chschlue — APPROVED** [review]
Thanks.

It's GH's review feature (available on the `Files changed` tab). I like it because it clearly separates concrete change requests from chit-chat.

**Pull request merged**